internal/server/server.go

```diff
@@ -551,6 +553,35 @@ func (s *Server) VerifyIdentityHeader(nextHandler echo.HandlerFunc) echo.Handler
 	return func(ctx echo.Context) error {
 		request := ctx.Request()
 
+		loader := openapi3.NewSwaggerLoader()
+		doc, err := loader.LoadSwaggerFromFile("api.yaml")
+		if err != nil {
+			panic(err)
+		}
+
+		if err := doc.Validate(loader.Context); err != nil {
+			panic(err)
+		}
+
+		router, err := legacyrouter.NewRouter(doc)
+		if err != nil {
+			panic(err)
+		}
+		route, params, err := router.FindRoute(request)
+		if err != nil {
+			panic(err)
```

Comment:
Better not to use panic here, juts return the error. The Errorhandler will turn it into a 500, with panic the server will crash when handling requests. (Inside of `NewServer` it's fine to panic however, if anything goes wrong there we want to crash the server).